cmd/gateway/main.go

```diff
@@ -71,6 +72,7 @@ func init() {
 	configFlags.String("cb-host", "localhost", "the couchbase server host")
 	configFlags.String("cb-user", "Administrator", "the couchbase server username")
 	configFlags.String("cb-pass", "password", "the couchbase server password")
+	configFlags.Bool("cb-auth-from-awssm", false, "fetch password from aws ssm")
```

Comment:
I would say this should probably be a string which specifies what secret name to pull.

Reply:
I think the region might also need to be something that can be specified as an option.

Reply:
The secret name uses the cluster ID so that'd be awkward to get in the service file to pass in as a flag. 

I deployed a cluster in us-west-2 and CNG fetched the secret and connected, but I'll check with Shelby about it.

Reply:
I suspect the system (dp-agent-cng?) that's providing the options for starting cng will probably know the cluster-id, so it should be able to trivially generate it.

Reply:
How would we then provide that to cng without having dp-agent start CNG? Which is what we are trying to avoid. From what I understood from Shelby is that he wants CNG to start independently from dp-agent, without dp-agent having to do anything.

Reply:
I think while we are trying to avoid using `dp-agent` itself, there will ultimately need to be some tool in the middle between CNG and Capella in order to do other aspects of the integration.  Shelby and I have discussed potentially still using "disk" (but an in-memory file system) to pass the certificates between Capella and CNG, and there are some aspects of CNG that will need to be reconfigured while it's operational through our file watching.  These pieces will likely need to be managed by some sort of agent component.